fsd_utils/locale_selector/set_lang.py

```diff
@@ -33,10 +35,12 @@ def __init__(self, app):
 
     @staticmethod
     def select_language(locale):
-
-        response = make_response(
-            redirect(request.referrer or request.args.get("return_url") or "/", 302)
-        )
+        referrer = request.referrer or "/"
+        referrer = referrer.replace("\\", "/")
+        if not urlparse(referrer).netloc and not urlparse(referrer).scheme:
```

Comment:
It would be good to run this through a few scenarios of what we'd expect to be referred and what it should default to `/` for but appreciate this wants to be in sooner than later

Reply:
@sfount added some tests in b70127d